homeassistant/components/subaru/__init__.py

```diff
@@ -0,0 +1,189 @@
+"""The Subaru integration."""
+import asyncio
+from datetime import timedelta
+import logging
+import time
+
+from subarulink import Controller as SubaruAPI, SubaruException
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_DEVICE_ID, CONF_PASSWORD, CONF_PIN, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers import aiohttp_client, config_validation as cv
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import (
+    CONF_COUNTRY,
+    CONF_UPDATE_ENABLED,
+    COORDINATOR_NAME,
+    DOMAIN,
+    ENTRY_CONTROLLER,
+    ENTRY_COORDINATOR,
+    ENTRY_VEHICLES,
+    FETCH_INTERVAL,
+    REMOTE_SERVICE_FETCH,
+    SUPPORTED_PLATFORMS,
+    UPDATE_INTERVAL,
+    VEHICLE_API_GEN,
+    VEHICLE_HAS_EV,
+    VEHICLE_HAS_REMOTE_SERVICE,
+    VEHICLE_HAS_REMOTE_START,
+    VEHICLE_HAS_SAFETY_SERVICE,
+    VEHICLE_LAST_UPDATE,
+    VEHICLE_NAME,
+    VEHICLE_VIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+REMOTE_SERVICE_SCHEMA = vol.Schema({vol.Required(VEHICLE_VIN): cv.string})
+
+
+async def async_setup(hass, base_config):
+    """Do nothing since this integration does not support configuration.yml setup."""
+    hass.data.setdefault(DOMAIN, {})
+    return True
+
+
+async def async_setup_entry(hass, entry):
+    """Set up Subaru from a config entry."""
+    config = entry.data
+    websession = aiohttp_client.async_get_clientsession(hass)
+    try:
+        controller = SubaruAPI(
+            websession,
+            config[CONF_USERNAME],
+            config[CONF_PASSWORD],
+            config[CONF_DEVICE_ID],
+            config[CONF_PIN],
+            None,
+            config[CONF_COUNTRY],
+            update_interval=UPDATE_INTERVAL,
+            fetch_interval=FETCH_INTERVAL,
+        )
+        _LOGGER.debug("Using subarulink %s", controller.version)
+        await controller.connect()
+    except SubaruException as err:
+        raise ConfigEntryNotReady(err.message) from err
```

Comment:
If the Credentials are invalid, we should log an error and return false￼ As we only want to retry when the service is not available or having issues￼